app/controllers/concerns/user_invitation.rb

```diff
@@ -62,6 +72,22 @@ def invite_new_user(email:, login: nil, first_name: nil, last_name: nil)
   end
 
   ##
+  # Sends a new invitation to the user with a new token.
+  #
+  # @param user_id [Integer] ID of the user to be re-invited.
+  # @return [Token] The new token used for the invitation.
+  def reinvite_user(user_id)
+    token = Token.find_by user_id: user_id, action: token_action
+
+    token.destroy! if token
+    new_token = Token.create! user_id: user_id, action: token_action
```

Comment:
I don't even know what the actual error will be myself in this case. In 99%
of the cases this will just work. There will be no validation errors. The
only error that can occur is some unknown database error preventing the
record from being saved. So it's fine if the user is clueless because I am
too IMO. I wouldn't want to display an arbitrary technical error message in
this case.

2016-02-02 9:26 GMT+00:00 Oliver Günther <notifications@github.com>:

> In app/controllers/concerns/user_invitation.rb
> <https://github.com/opf/openproject/pull/4100#discussion_r51542659>:
>
> > @@ -62,6 +72,22 @@ def invite_new_user(email:, login: nil, first_name: nil, last_name: nil)
> >    end
> >
> >    ##
> > +  # Sends a new invitation to the user with a new token.
> > +  #
> > +  # @param user_id [Integer] ID of the user to be re-invited.
> > +  # @return [Token] The new token used for the invitation.
> > +  def reinvite_user(user_id)
> > +    token = Token.find_by user_id: user_id, action: token_action
> > +
> > +    token.destroy! if token
> > +    new_token = Token.create! user_id: user_id, action: token_action
>
> But raising an internal error will leave the user clueless of the actual
> error in production mode. That can't be what you want?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/opf/openproject/pull/4100/files#r51542659>.
>